toolkit/primitives/address-associations/src/lib.rs

```diff
@@ -0,0 +1,21 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use parity_scale_codec::Encode;
+use sidechain_domain::{MainchainAddress, UtxoId};
+
+#[derive(Debug, Clone, Encode)]
+pub struct AddressAssociationSignedMessage<SCAddr> {
+	pub mainchain_address: MainchainAddress,
+	pub partnerchain_address: SCAddr,
+	pub genesis_utxo: UtxoId,
+}
+
+sp_api::decl_runtime_apis! {
```

Comment:
please add this API to the runtime later

Reply:
also we should rather avoid providing new RuntimeApi's, unless it is really needed